### PR TITLE
Add transaction application, finish migration script and admin page

### LINKED ISSUE #15: Multimedia Ordering (Cart and Transactions)
- [x] migration script (CREATE & DROP)
- [x] View
- [x] Delete Cart Item
- [x] Checkout Item

### PR DESCRIPTION
**Pull request by mikelimantara**
Fixes #15

### CODE REVIEW

File: transaction/models.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    class Meta:
+        managed = False
+        db_table = 'cart'
+
+
+class CartItem(models.Model):
+    id = models.AutoField(primary_key=True)
+    cart = models.ForeignKey('Cart')
+    object = models.ForeignKey(multimedia.models.Multimedia)
```
Comment (9gix) [inline transaction/models.py]: @mikelimantara avoid overwriting the `object` of python global variable.

File: transaction/models.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    object_type = models.CharField(max_length=45)
```
Comment (9gix) [inline transaction/models.py]: I hope the Django ContentType could help you in the design of this generic type
https://docs.djangoproject.com/en/1.7/ref/contrib/contenttypes/

**Comment by 9gix** [comment]
@mikelimantara change the editor to use 4 space instead of tab character.

**Comment by digawp** [comment]
Actually we just discussed ytd, having a quantity in our case is weird, because we are a digital content store, why would you buy more than 1 of a digital content?

**Comment by 9gix** [comment]
@digawp @mikelimantara How about dropping the quantity field since it does not provide additional value?

**Comment by mikelimantara** [comment]
@digawp @9gix I actually agree to drop the quantity field since users are unlikely to buy a multimedia for a quantity of more than 1. And, I think it is possible to use aggregate for the sum.

**Comment by digawp** [comment]
+1 dropping quantity field

File: dcs/urls.py [hunk 3]
```diff
@@ -10,9 +10,11 @@
     # url(r'^blog/', include('blog.urls')),
 
     url(r'^$', 'dcs.views.home', name='home'), # mock, CHANGE later!
+    url(r'^nimda/', include('grappelli.urls')),
     url(r'^admin/', include(admin.site.urls)),
     url(r'^accounts/', include('registration.backends.default.urls')),
     url(r'^', include('multimedia.urls', namespace='multimedia')),
+    url(r'^carts/', include('transaction.urls')),
```
Comment (digawp) [inline dcs/urls.py]: I think this line needs to come before the multimedia url routing, but I'm not sure. But regardless, don't forget to namespace it later

**Comment by mikelimantara** [comment]
Question: @digawp @9gix 
Can we add a cart when user first registers or completed transaction? This is so that at any point of time, user always have an empty cart (I think need this cart_id in the url).

By doing this:
1) when user clicks on add to cart: simply add item to the previously created cart (select user's cart which is created last - probably use 'created at' field)
2) when user clicks purchase, insert a new empty cart for this user, and also, insert a new entry to transaction table specifying the id of the cart purchased

**Comment by mikelimantara** [comment]
@digawp does the order of the url matter in this case? Because I can still access book and music in my localhost

**Comment by digawp** [comment]
I'm okay with the creating cart issue (I think that should be posted at the issue not here but whatev)

Regarding order of URL, I don't know, that's why I asked haha. B/c if Django does the url lookup linearly, then any url will be routed to multimedia before it reaches the cart routing. If Django does URL lookup by longest matching prefix, then the order does not matter. Again, your reply should be on the line itself (so that it is easy to follow a convo thread), but again, whatev.

File: transaction/views.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    __deleteItem(cart_id, cart_item_id)
+
+    return redirect("/carts/")
+
+def __getUserCartId(user_id):
+    cart_raw = Cart.objects.raw("""SELECT c1.id 
+                                   FROM cart c1
+                                   WHERE c1.buyer_id = %s AND
+                                         c1.is_completed = false AND
+                                         c1.created_at >= ALL (SELECT c2.created_at FROM cart c2);
```
Comment (digawp) [inline transaction/views.py]: sql indent 2 lines better

File: transaction/views.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponseRedirect
+from django.db import connection
+
+import ipdb
```
Comment (9gix) [inline transaction/views.py]: Remove ipdb when committing

File: transaction/views.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    __deleteItem(cart_id, cart_item_id)
+
+    return redirect("/carts/")
+
+def __getUserCartId(user_id):
+    cart_raw = Cart.objects.raw("""SELECT c1.id 
+                                   FROM cart c1
+                                   WHERE c1.buyer_id = %s AND
+                                         c1.is_completed = false AND
+                                         c1.created_at >= ALL (SELECT c2.created_at FROM cart c2);
```
Comment (digawp) [inline transaction/views.py:74]: Bressan -> :+1:

**Pull request merged**